crates/bevy_ecs/src/query/access.rs

```diff
@@ -325,6 +354,57 @@ impl<T: SparseSetIndex> FilteredAccess<T> {
     }
 }
 
+// A struct to express something like `Or<(With<A>, With<B>)>`.
+// Filters like `(With<A>, Or<(With<B>, With<C>)>` are expanded into `Or<((With<A>, With<B>), (With<A>, With<C>))>`.
+#[derive(Clone, Eq, PartialEq)]
+struct ExpandedOrWithAccesses {
```

Comment:
I think this would be clearer if it was just `WithAccesses`, and explained further about `Or` behavior in the doc strings.

Reply:
(the implementation evolved a bit, and so did the comments, resolving it in favour of comments to the up-to-date diff)